### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by dwhswenson**
Some cleanup for docs and some new materials added.

* Removed the "Guides" heading (as separate from "Topics") and moved everything into "Topics". It wasn't clear to me what the distinction was supposed to be, and I'm pretty sure it wouldn't be clear to users. I also don't show a couple things that I don't think were really meant to be for users (like levels of user experience -- that was supposed to be internal for discussion purposes)
* New docs for data objects (some of which taken from the current draft of the paper)
* Lots of little cleanup to fix warnings that sphinx was throwing.
* Removed the "Full API" link, which wasn't really full, and was raising warnings in Sphinx. Maybe we should look at `sphinx-apidoc` and see if we can get actual complete API documentation from that?
* Automated the docs versioning using `pkg_resources` and `packaging.version`. Requires installing OPS before building docs, but I don't think that's a problem (needs to be installed to build anyway, doesn't it?) Makes one less place to update version numbers.

### CODE REVIEW

**Comment by dwhswenson** [comment]
Passes test. Ready for review.

**Comment by dwhswenson** [comment]
Could this get a review ASAP? I'll be telling people about OPS at the "Recent Advances in Modeling Rare Events" conference at the end of this week, and I'd rather point them to the website *after* these changes have been made.

File: docs/topics/absolute_beginners.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+conceptual framework that OpenPathSampling is built on.
+
+Path sampling techniques apply, obviously, to "paths." We will use the word
+"trajectory" interchangeably with "path." Each path is made up of several
+"snapshots" or "frames." Again, we often use these words interchangeably.
+
+Collective variables are functions of a snapshot
+================================================
+
+For a system with :math:`N` atoms, each snapshot on a trajectory (in 3D
```
Comment (jhprinz) [inline docs/topics/absolute_beginners.rst]: We might explain the use of _snapshot_ as a frozen picture of the system state at a particular in time (which is more or can be more than just a phase space point, etc...)

File: docs/topics/absolute_beginners.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+Since the collective variable takes a snapshot and maps it to a single
+number, we can determine whether any snapshot is within that volume.
+
+Ensembles apply to trajectories
+===============================
+
+We also have a class of objects called :class:`.Ensemble`. More correctly,
+we should call these "path ensembles" (or even more correctly, "path
+ensemble functions.") The most important function this provides is the
+indicator function, which tells you whether or a given trajectory is in the
```
Comment (jhprinz) [inline docs/topics/absolute_beginners.rst]: remove _or_

File: docs/topics/ensembles_apply_to_trajectories_not_frames.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+It is common to initially think of ensembles as applying on a frame-by-frame
+basis. However, ensembles are only valid on trajectories, not individual
+frames. To explain this, let’s consider a couple simple common misconceptions.
+
+Combining volume ensembles doesn’t combine volumes
+==================================================
+
+
+In might seem like ``Ensemble(volume_A) & Ensemble(volume_B)`` should be the
```
Comment (jhprinz) [inline docs/topics/ensembles_apply_to_trajectories_not_frames.rst]: It ...

**Comment by dwhswenson** [comment]
Fixed those typos. Also removed the wrapping of `.md` files in `.rst`, done in `premake.py`, since I converted the Markdown to `.rst`. Going to merge this now, since the changes were approved!

**Pull request merged**